packages/core/src/common/command.ts

```diff
@@ -273,6 +340,52 @@ export class CommandRegistry implements CommandService {
         return typeof this.getVisibleHandler(command, ...args) !== 'undefined';
     }
 
+    /**
+     * Comments for trackEnabled also apply here.
+     */
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    trackVisible(commandId: string, ...args: any[]): HandlerPropertyTracker {
```

Comment:
Yes, it was a lot of duplicated code.  I had originally left it duplicated because I was not sure the same logic should be in both.  The two properties are not really independent but interact with each other.

I had been thinking that really we should have a single three-way state.  Why I replaced the two booleans with Active/Disabled/Hidden it simplified quite a bit.  So I have updated the PR accordingly.  `trackVisible` and `trackEnabled` are now replaced by `trackActiveState`.  I would be interested to know if you feel this is a better approach.